src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -81,6 +81,7 @@ public async Task<IActionResult> Delete(Guid instanceGuid, Guid dataId, int inst
                 return NotFound("Provided instanceId is unknown to storage service");
             }
 
+            _dataRepository.SetAppBlobClient(instance.Org);
```

Comment:
Should we introduce a context instead? For instance:
`using (var dataContext = _dataRepository.GetOrgContext(instance.Org)`
where the dataContext is an idisposable and unsets the appcontainer when it is disposed? If we are forgetting to set the app context, then we may overwrite data stored at other orgs containers due to previous set app container.